### PR TITLE
Do Not Sell MVP

### PR DESCRIPTION
**Pull request by marcosholgado**
Task/Issue URL: https://app.asana.com/0/inbox/1125189844075764/1191298152080291/1196344248236896
Tech Design URL: 
CC: 

**Description**:
This PR adds a new setting to enable/disable Global Privacy Control (aka Do Not Sell)

Note: We are using out typical settings colors rather than ones from the design.

**Steps to test this PR**:

**Activate signal** 
1. Navigate to `https://global-privacy-control.glitch.me/` (it needs to be a new URL load, refresh or navigate back won't work)
1. Server side detection and client side detection should show a thumb down icon in red.
1. Go to settings.
1. Global Privacy Control should have a "Disabled" subtitle.
1. Go to Global Privacy Control.
1. Pixel `ms_dns` should be fired.
1. Enable the toggle.
1. Pixel `ms_dns_on` should be fired.
1. Navigate to `https://global-privacy-control.glitch.me/`
1. Server side detection and client side detection should show a thumb up icon in blue.

**Deactivate signal**
1. Following the above scenario, go to settings.
1. Global Privacy Control should have an "Enabled" subtitle.
1. Go to Global Privacy Control.
1. Pixel `ms_dns` should be fired.
1. Disable the toggle.
1. Pixel `ms_dns_off` should be fired.
1. Navigate to `https://global-privacy-control.glitch.me/` (it needs to be a new URL load, refresh or navigate back won't work)
1. Server side detection and client side detection should show a thumb down icon in red.

**Learn more**
1. Go to settings and Global Privacy Control
1. Below the last paragraph there should be a `Learn More` link, click on it.
1. You should be taken to `https://duckduckgo.com/global-privacy-control-learn-more` which will redirect into `https://spreadprivacy.com/announcing-global-privacy-control/`

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Review by malmstein — CHANGES_REQUESTED** [review]
Something seems off in the UI @marcosholgado. The padding between elements seems bigger that the one we use in similar screens:

| Fireproof  | GPC |
| ------ | ----- | 
device-2020-10-01-115049|device-2020-10-01-115132|

Ideally we'd have the same padding in all the screens that have the same look and feel. Location and Fireproof sites follow this pattern.

**Comment by marcosholgado** [comment]
> Something seems off in the UI @marcosholgado. The padding between elements seems bigger that the one we use in similar screens:
> 
> Ideally we'd have the same padding in all the screens that have the same look and feel. Location and Fireproof sites follow this pattern.

Nice catch! Should be good now!

File: app/src/androidTest/java/com/duckduckgo/app/globalprivacycontrol/GlobalPrivacyControlInjectorJsTest.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    @UiThreadTest
+    @Test
+    fun whenInjectDoNotSellToDomAndGcpIsEnabledThenInjectToDom() {
+        val webView = spy(WebView(InstrumentationRegistry.getInstrumentation().targetContext))
+        whenever(mockSettingsStore.globalPrivacyControlEnabled).thenReturn(true)
+
+        testee.injectDoNotSellToDom(webView)
+
+        verify(webView).evaluateJavascript(any(), anyOrNull())
```
Comment (malmstein) [inline app/src/androidTest/java/com/duckduckgo/app/globalprivacycontrol/GlobalPrivacyControlInjectorJsTest.kt:51]: isn't there a way to check that the javascript has properly been evaluated? verifying null seems odd

File: app/src/androidTest/java/com/duckduckgo/app/globalprivacycontrol/GlobalPrivacyControlInjectorJsTest.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    @UiThreadTest
+    @Test
+    fun whenInjectDoNotSellToDomAndGcpIsNotEnabledThenDoNotInjectToDom() {
+        val webView = spy(WebView(InstrumentationRegistry.getInstrumentation().targetContext))
+        whenever(mockSettingsStore.globalPrivacyControlEnabled).thenReturn(false)
+
+        testee.injectDoNotSellToDom(webView)
+
+        verifyZeroInteractions(webView)
```
Comment (malmstein) [inline app/src/androidTest/java/com/duckduckgo/app/globalprivacycontrol/GlobalPrivacyControlInjectorJsTest.kt:62]: same here, this doesn't really test that the javascript has not been injected but that there are no interactions with the webview

File: app/src/androidTest/java/com/duckduckgo/app/globalprivacycontrol/GlobalPrivacyControlInjectorJsTest.kt (hunk 1 above)
Comment (marcosholgado) [reply app/src/androidTest/java/com/duckduckgo/app/globalprivacycontrol/GlobalPrivacyControlInjectorJsTest.kt]: Sure thing, done!

File: app/src/androidTest/java/com/duckduckgo/app/globalprivacycontrol/GlobalPrivacyControlInjectorJsTest.kt (hunk 2 above)
Comment (marcosholgado) [reply app/src/androidTest/java/com/duckduckgo/app/globalprivacycontrol/GlobalPrivacyControlInjectorJsTest.kt]: And also done!

**Review by malmstein — APPROVED** [review]
Everything works as expected, great job @marcosholgado

**Pull request merged**